File: selector/src/main/java/com/luck/picture/lib/adapter/holder/ImageViewHolder.java
package com.luck.picture.lib.adapter.holder;

import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.luck.picture.lib.R;
import com.luck.picture.lib.config.PictureMimeType;
import com.luck.picture.lib.config.SelectorConfig;
import com.luck.picture.lib.config.SelectorProviders;
import com.luck.picture.lib.entity.LocalMedia;
import com.luck.picture.lib.style.SelectMainStyle;
import com.luck.picture.lib.utils.MediaUtils;
import com.luck.picture.lib.utils.StyleUtils;

/**
 * @author：luck
 * @date：2021/11/20 3:59 下午
 * @describe：ImageViewHolder
 */
public class ImageViewHolder extends BaseRecyclerMediaHolder {
    private final ImageView ivEditor;
    private final TextView tvMediaTag;

    public ImageViewHolder(View itemView, SelectorConfig config) {
        super(itemView, config);
        tvMediaTag = itemView.findViewById(R.id.tv_media_tag);
        ivEditor = itemView.findViewById(R.id.ivEditor);
        SelectMainStyle adapterStyle = selectorConfig.selectorStyle.getSelectMainStyle();
        int imageEditorRes = adapterStyle.getAdapterImageEditorResources();
        if (StyleUtils.checkStyleValidity(imageEditorRes)) {
            ivEditor.setImageResource(imageEditorRes);
        }
        int[] editorGravity = adapterStyle.getAdapterImageEditorGravity();
        if (StyleUtils.checkArrayValidity(editorGravity)) {
            if (ivEditor.getLayoutParams() instanceof RelativeLayout.LayoutParams) {
                ((RelativeLayout.LayoutParams) ivEditor.getLayoutParams()).removeRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
                for (int i : editorGravity) {
                    ((RelativeLayout.LayoutParams) ivEditor.getLayoutParams()).addRule(i);
                }
            }
        }

        int[] tagGravity = adapterStyle.getAdapterTagGravity();
        if (StyleUtils.checkArrayValidity(tagGravity)) {
            if (tvMediaTag.getLayoutParams() instanceof RelativeLayout.LayoutParams) {
                ((RelativeLayout.LayoutParams) tvMediaTag.getLayoutParams()).removeRule(RelativeLayout.ALIGN_PARENT_END);
                ((RelativeLayout.LayoutParams) tvMediaTag.getLayoutParams()).removeRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
                for (int i : tagGravity) {
                    ((RelativeLayout.LayoutParams) tvMediaTag.getLayoutParams()).addRule(i);
                }
            }
        }
        int background = adapterStyle.getAdapterTagBackgroundResources();
        if (StyleUtils.checkStyleValidity(background)) {
            tvMediaTag.setBackgroundResource(background);
        }

        int textSize = adapterStyle.getAdapterTagTextSize();
        if (StyleUtils.checkSizeValidity(textSize)) {
            tvMediaTag.setTextSize(textSize);
        }

        int textColor = adapterStyle.getAdapterTagTextColor();
        if (StyleUtils.checkStyleValidity(textColor)) {
            tvMediaTag.setTextColor(textColor);
        }
    }


    @Override
    public void bindData(LocalMedia media, int position) {
        super.bindData(media, position);
        if (media.isEditorImage() && media.isCut()) {
            ivEditor.setVisibility(View.VISIBLE);
        } else {
            ivEditor.setVisibility(View.GONE);
        }
        tvMediaTag.setVisibility(View.VISIBLE);
        if (PictureMimeType.isHasGif(media.getMimeType())) {
            tvMediaTag.setText(mContext.getString(R.string.ps_gif_tag));
        } else if (PictureMimeType.isHasWebp(media.getMimeType())) {
            tvMediaTag.setText(mContext.getString(R.string.ps_webp_tag));
        } else if (MediaUtils.isLongImage(media.getWidth(), media.getHeight())) {
            tvMediaTag.setText(mContext.getString(R.string.ps_long_chart));
        } else {
            tvMediaTag.setVisibility(View.GONE);
        }
    }
}

File: selector/src/main/java/com/luck/picture/lib/adapter/ImageViewHolder.kt
package com.luck.picture.lib.adapter

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.luck.picture.lib.R
import com.luck.picture.lib.entity.LocalMedia
import com.luck.picture.lib.utils.MediaUtils

/**
 * @author：luck
 * @date：2022/11/30 3:33 下午
 * @describe：ImageViewHolder
 */
open class ImageViewHolder(itemView: View) : ListMediaViewHolder(itemView) {
    private var tvMediaTag: TextView = itemView.findViewById(R.id.tv_media_tag)
    private var ivEditor: ImageView = itemView.findViewById(R.id.iv_editor)

    override fun bindData(media: LocalMedia, position: Int) {
        super.bindData(media, position)
        onMergeEditor(media)
        when {
            isLongImage(media.width, media.height) -> {
                tvMediaTag.visibility = View.VISIBLE
                val longChartTag = itemView.context.getString(R.string.ps_long_chart)
                when {
                    isGif(media.mimeType, media.getAvailablePath()) -> {
                        tvMediaTag.text =
                            String.format("$longChartTag,${itemView.context.getString(R.string.ps_gif_tag)}")
                    }
                    isWebp(media.mimeType, media.getAvailablePath()) -> {
                        tvMediaTag.text =
                            String.format("$longChartTag,${itemView.context.getString(R.string.ps_webp_tag)}")
                    }
                    else -> {
                        tvMediaTag.text = longChartTag
                    }
                }
            }
            isGif(media.mimeType, media.getAvailablePath()) -> {
                tvMediaTag.text = itemView.context.getString(R.string.ps_gif_tag)
                tvMediaTag.visibility = View.VISIBLE
            }
            isWebp(media.mimeType, media.getAvailablePath()) -> {
                tvMediaTag.text = itemView.context.getString(R.string.ps_webp_tag)
                tvMediaTag.visibility = View.VISIBLE
            }
            else -> {
                tvMediaTag.visibility = View.GONE
            }
        }
    }

    open fun isWebp(mimeType: String?, path: String?): Boolean {
        return MediaUtils.isHasWebp(mimeType) || MediaUtils.isUrlHasWebp(path)
    }

    open fun isGif(mimeType: String?, path: String?): Boolean {
        return MediaUtils.isHasGif(mimeType) || MediaUtils.isUrlHasGif(path)
    }

    open fun isLongImage(width: Int, height: Int): Boolean {
        return MediaUtils.isLongImage(width, height)
    }

    open fun onMergeEditor(media: LocalMedia) {
        val selectResult = mGetSelectResultListener?.onSelectResult()
        if (selectResult != null && selectResult.isNotEmpty()) {
            if (!media.isEditor()) {
                val position = selectResult.indexOf(media)
                if (position >= 0) {
                    val existsMedia = selectResult[position]
                    if (existsMedia.isEditor()) {
                        media.cropWidth = existsMedia.cropWidth
                        media.cropHeight = existsMedia.cropHeight
                        media.editorPath = existsMedia.editorPath
                        media.editorData = existsMedia.editorData
                        media.cropOffsetX = existsMedia.cropOffsetX
                        media.cropOffsetY = existsMedia.cropOffsetY
                        media.cropAspectRatio = existsMedia.cropAspectRatio
                    }
                }
            }
        }
        ivEditor.visibility = if (media.isEditor()) View.VISIBLE else View.GONE
    }
}